### ISSUE TITLE
Add org-mode support

### ISSUE THREAD

**Issue by vHanda** [GitJournal/GitJournal#145]
I've never really used org-mode, but it's something many developers seem to love. GitJournal could support org-mode files, and not just markdown.

---
> Please vote on this issue with a 👍reaction. It's one of the primary mechanisms used to prioritize issues.

**Comment by vHanda** [comment]
There seems to be a read-only org-mode parser I could use - https://github.com/amake/org_parser

**Comment by 0atman** [comment]
Absolutely love to see this, the first version might just treat them as plain text files, no special treatment needed!

**Comment by MisterY** [comment]
That's a nice parser, or rather the Orgro app that implements it. With the parser it would be really easy to start with .org files and display them correctly.
In that sense, you may want to split this task into many smaller ones. Displaying the .org files would already be very useful in itself. Hopefully that can be done rather quickly.

Editing files, and applying other org-mode features (like time tracking, progress updates, progress tracking, etc.), will take much longer. Fortunately, there are plans for much of that functionality already noted here in the repo.

**Comment by 0atman** [comment]
Agreed, just treat them as .txt to start off with, anything else can be prioritised later.
I don't tend to write much on mobile, but an edit here or there would be amazing

**Comment by MisterY** [comment]
After having used Orgro for a short while, I believe that supporting orgmode at a functionalllevel should be fairly easy if separate editor screens are used.
For example, when a title is selected and the Edit button clicked, it should open just that item. This is just how Orgzly does it and allows supporting files of unlimited size and levels.
The editor may be plain text. The parser will correctly display it in the preview mode.
Alternatively, a toolbar similar to the Markdown toolbar, could be implemented in a plain text editor, allowing indentation, check boxes and other items that the Markdown toolbar already has.

**Comment by MisterY** [comment]
The initial version can now be considered complete. This covers the basics - .org files editing (as plain text), preview, image display, and link handling.
Any further improvements can be done incrementally from here.

**Comment by 0atman** [comment]
Fantastic!

**Comment by alzxjm** [comment]
> The initial version can now be considered complete. This covers the basics - .org files editing (as plain text), preview, image display, and link handling.
> Any further improvements can be done incrementally from here.

I might be missing something, but images aren't quite right for me. For one, using the org editor and attaching a photo uses Markdown syntax rather than org syntax and for two, the preview mode for the org editor doesn't show any images inline.

**Comment by vHanda** [comment]
@MisterY : Could you please take a look at this commit for fixing the image markup?

@alzxjm : You're right, at least based on the code, I still haven't tested out org-mode.

I'm leaving this issue open until one can create new files which are in orgMode. This requires some refactoring as GitJournal was originally only meant for markdown files.

**Comment by MisterY** [comment]
@vHanda, thanks for the ping. I'll have a look in the coming days.

**Comment by alzxjm** [comment]
Adding this much org support isn't too much feature creep, is it? I guess it's just another markup language and it's pretty standard, but org-mode is a big beast. 

It'd be awesome to have, don't get me wrong. Having Beorg for the more agenda/task parts of org-mode and GitJournal for editing org files would be a dream come true. 

I'm happy to test out features, as well.

**Comment by MisterY** [comment]
@vHanda, yes, this may go in. Certainly better than the MD images. I wanted to check the changes and update the Org parser, as there have been improvements in handling images. If someone has the time, that would be great, as well.

@alzxjm, my understanding is that Org support will come, sooner or later, so it's not a feature creep. But there's some infrastructure work that needs to be done first in order to properly work on the Org editor. So, currently the editor is a bit limited but works as a pure text, which is ok. The parser and the display is fairly good and complete.
